openverse_catalog/dags/providers/provider_api_scripts/rawpixel.py

```diff
@@ -1,193 +1,282 @@
+            'image_png_100', 'image_150', 'image_png_150', 'image_200', 'image_png_200',
+            'image_250', 'image_png_250', 'image_300', 'image_png_300', 'image_400',
+            'image_png_400', 'image_500', 'image_png_500', 'image_600', 'image_png_600',
+            'image_700', 'image_png_700', 'image_800', 'image_png_800', 'image_900',
+            'image_png_900', 'image_1000', 'image_png_1000', 'image_1300',
+            'image_png_1300'
+
+        The number refers to the width displayed, and a png option is provided for each
+        size.
+        """
+        style_uri = data.get("style_uri")
+        if not style_uri:
+            return None
+        return style_uri.format(size_option)
+
+    @staticmethod
+    def _get_image_properties(data: dict) -> tuple[None, None] | tuple[int, int]:
+        width = max(data.get("width", 0), data.get("display_image_width", 0))
+        height = max(data.get("height", 0), data.get("display_image_height", 0))
+        if (width, height) == (0, 0):
+            return None, None
+        return width, height
+
+    @staticmethod
+    def _clean_text(text: str) -> str:
+        # First clear full patterns
+        text = RawpixelDataIngester.suffix_pattern_full.sub("", text)
+        # Then clear partial patterns
+        text = RawpixelDataIngester.suffix_pattern_partial.sub("", text)
+        # Unescape HTMl sequences
+        text = html.unescape(text)
+        # Clean whitespace
+        text = text.strip()
+        return text
+
+    @staticmethod
+    def _get_title(metadata: dict) -> str | None:
+        """
+        Titles come in the following form, so we clean them up a bit:
+          Bull elk searches for food | Free Photo - rawpixel
+          Desktop wallpaper summer beach landscape, | Free Photo - rawpixel
+          Branch with a sunflower (1714&ndash;1760) | Free Photo Illustration - rawpixel
+          Free public domain CC0 photo. | Free Photo - rawpixel
+          Flower background. Free public domain | Free Photo - rawpixel
+        """
+        title: str = metadata.get("title")
+        if not title:
+            return None
+        # Cut off the last bit about Rawpixel
+        title = title.split("|", maxsplit=1)[0].strip()
+        # Clean text
+        title = RawpixelDataIngester._clean_text(title)
+        # Remove trailing punctuation
+        title = title.removesuffix(",").removesuffix(".")
+        return title or None
+
+    @staticmethod
+    def _get_meta_data(data: dict, metadata: dict) -> dict:
+        description = RawpixelDataIngester._clean_text(
+            metadata.get("description_text") or ""
+        )
+        meta_data = {
+            "description": description or None,
+            "download_count": data.get("download_count"),
+            "thumbnail_url": RawpixelDataIngester._get_image_url(
+                data, RawpixelDataIngester.thumbnail_size_option
+            ),
```

Comment:
This should go in the thumbnails columns.
```suggestion
```

Reply:
Since the actual thumbnail column is being overridden to `None` in the MediaStore, I'd prefer we _either_ keep this in `meta_data`, where it will not be overridden and as set by recent precedent, or just not include it at all if we don't think we need this information for RawPixel.

Most of the discussion so far about what to do with `thumbnail_url` is on #698. I believe the intention is to eventually remove the `thumbnail_url` column, since it is probably not worth keeping an entire column only used by a handful of providers, and use the `meta_data` instead (but sparingly, when we have a good reason to supply the thumbnail separately).

To be fair I don't think that's been "officially" decided, but we should make sure we come to an agreement about this since it's now affecting a handful of providers!

Reply:
This is why it will be nice to tackle the "Data normalization" milestone sooner than later, so we have the same general understanding of what to put in each column, but given it needs an agreement, as you mention, that is why I suggested omitting it here in the refactoring. We can create another issue for the Rawpixel thumbnails.

The discussion has occurred sparsely through many issues so [will be moved to a Make blog post](https://github.com/WordPress/openverse-catalog/pull/796#issuecomment-1287422102) to make a decision.

Reply:
Okay, I'll remove the actual insertion code but leave in the other machinery for when we go back and add these eventually. Issue created here: https://github.com/WordPress/openverse-catalog/issues/817